spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -779,10 +800,54 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
             Object value = CatalystTypeConverters.convertToScala(catalystValue, field.dataType());
             values.put(field.name(), value.toString());
           });
-          return new SparkPartition(values, partition.path().toString(), format);
+          FileStatus fileStatus =
+              scala.collection.JavaConverters.seqAsJavaListConverter(partition.files()).asJava().get(0);
+
+          return new SparkPartition(values, fileStatus.getPath().getParent().toString(), format);
         }).collect(Collectors.toList());
   }
 
+  private static List getPartitionFilterExpressions(StructType schema,
+                                                    Map<String, String> partitionFilter) {
+    List<org.apache.spark.sql.catalyst.expressions.Expression> filterExpressions = new java.util.ArrayList<>();
+    for (Map.Entry<String, String> entry : partitionFilter.entrySet()) {
+      try {
+        // IllegalArgumentException is thrown if schema doesn't contain this entry,
```

Comment:
Just wondering, how does  Spark parse these. Is it just in the ParseExpressions code and then fed directly into here? I know in the RewriteDataProcedure we end up using the parser to do a parseExpression since then we don't have to deal with all of these internal transforms and such. I am wondering if a similar approach could work here?

Reply:
Spark doesn't wrap this with its internal Exceptions. It throws this `java.lang.IllegalArgumentException` directly if the schema doesn't contains the column.

Reply:
Sorry I meant this whole conversion into Spark Expression classes. 

I know in Spark this probably happens in the parser and I was wondering if we could do the same thing here like

spark.parser.parseExpression("x = y")
I guess somewhere it does some conversion into strings or from strings into the proper types? Or maybe it doesn't...

Reply:
I tried `spark.parser.parseExpression("dept = hr")`, we will get a `EqualTo` filter with left side `UnresolvedAttribute` dept and right side `UnresolvedAttribute` hr. I think in `RewriteDataFilesProcedure` we will execute and the `UnresolvedAttribute` will get resolved. But in the case of `listFiles`, it doesn't go through the name resolution code and `UnresolvedAttribute` remains to be unresolved and cause problem.
I think I will have to manually construct this `EqualTo` filter. I will check more to see if I can find a better solution.

Reply:
I tried `spark.parser.parseExpression("value")`, hoping Spark can turn this into the right type, but it didn't work. I think I will have to construct the Literal and Filter Expression by myself.

Reply:
@huaxingao: As someone worked recently on the similar area. In my experience, It is better to avoid new implementation as it needs lot of testing (all data type * all expression) and chances of inducing issue is high.

For the unresolved expression issue with `spark.parser.parseExpression("value") `, I recently found a way to get the resolved expression by adding some prefix and collecting resolved expression from plan #3757  
May be can you see if it can help you as well ?

Reply:
@RussellSpitzer 
As @ajantha-bhat suggested, we can put the filter `"x = y"` inside a simple query and let Spark execute the query and then we collect the resolved filter Expression. This solution sounds good to me. WDYT?

Reply:
I like it, but do we have to change all the types then? I know spark gets upset if our types don't match column vs literal, and we have all string literals here :/

Reply:
Sorry for the late reply. I was off last Friday and yesterday. We don't need to change the types because Spark takes care of the casting.